osu.Game.Tournament/Screens/SetupScreen.cs

```diff
@@ -196,5 +200,56 @@ public bool Failing
                 }
             };
         }
+
+        private class ActionableInfoWithNumberBox : ActionableInfo
+        {
+            public new Action<int> Action;
+
+            private OsuNumberBox numberBox;
+
+            public int NumberValue
+            {
+                get
+                {
+                    int.TryParse(numberBox.Text, out var val);
+                    return val;
+                }
+                set => numberBox.Text = value.ToString();
+            }
+
+            protected override Drawable CreateComponent() => new Container
+            {
+                AutoSizeAxes = Axes.Y,
+                RelativeSizeAxes = Axes.X,
+                Children = new Drawable[]
+                {
+                    ValueText = new TournamentSpriteText
+                    {
+                        Anchor = Anchor.CentreLeft,
+                        Origin = Anchor.CentreLeft,
+                    },
+                    numberBox = new OsuNumberBox
+                    {
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.CentreRight,
+                        Width = 100,
+                        Margin = new MarginPadding
+                        {
+                            Right = 110
+                        }
```

Comment:
This is not a reliable way to do this sort of layouting. You'd want to nest this and the button in a fill flow:

```
Container
    ValueText (TournamentSpriteText)
    FillFlowContainer
        numberBox (OsuNumberBox)
        Button (TriangleButton)
```

Better yet, you could:

* add the fill flow to `ActionableInfo` and make `Button` its only child,
* expose the new fill flow as protected,
* and then here you wouldn't have to duplicate the body of `CreateComponent()`, but you could just call base and then add the number box to the exposed fill flow by calling
  ```
  Insert(0, numberBox = new OsuNumberBox { ... }
  ```

Reply:
I knew I was doing some hacky shit when coding this haha, thanks for the clear explanation, will implement it in the "better yet" way! 👍